### ISSUE TITLE
If user is logged out when posting a comment, the comment is still 'posted' as 'undefined'

### ISSUE THREAD

**Issue by lowercasename** [lowercasename/sweet#9]
There's an on-error Jquery which is meant to fire to prevent this, but it clearly needs fixing.

**Comment by toBeOfUse** [comment]
This is fixed in my latest commit. What was happening was that isLoggedIn() in routes.js was getting called, it was responding with a redirect to "/", the entire source code of the index page of the site was getting returned to the post request, and the check in scripts.js of "if(data)" was returning true because the entire source code of the index page is not a falsey value. I added a isLoggedInForPost() function that responds to post requests in an appropriate manner, as redirect responses make no sense in response to post requests. I think that this can be closed.

**Comment by toBeOfUse** [comment]
Note however that I did not switch any other app.post guys in routes.js to use my new isLoggedInForPost() as I thought some might depend on the old behavior of the isLoggedIn() function. And, I renamed the old isLoggedIn() to isLoggedInForGet(), as any function that responds with a redirect should really only be used for Get requests. So all the other app.post calls use isLoggedInForGet(), which looks weird. Might look into comprehensively changing this later.
